woocommerce/woocommerce
# Update Syncpack and use it to pin @wordpress packages to wp-6.0

### All Submissions:






### Changes proposed in this Pull Request:

There is a desire to sync all `@wordpress` packages in the repository to the same versions for consistency. We chose to match L2 as the version. Wordpress 6.2 will be released at the end of March which will put us at package version 6.0 for the next Wordpress release (7.6.0) so I'm pre-emptively setting the version to 6.0 here. Another reason I jumped to 6.0 is that the 5.9 packages are really looking old now and caused quite a few problems/conflicts. 

But, important note we don't need to pin `@wordpress` dependencies that are used as part of dev or testing because these dependencies are not exposed in Wordpress itself. This PR excludes those dependencies from being synchronized. At some point we may want to update and sync these dependencies (but not to `wp-6.0`) but it can be done one piece at a time because wholesale updating them could cause issues for teams that we haven't anticipated.

Other things of note:

* I updated syncpack to latest version. It has more informative output and fixes some oddities I saw during dev.

* Updating versions required fixing an issue with PostCSS version being used. I moved the postcss-loader to 4+ and with that you need to depend on postcss directly as well, I opted for 8+ which is already a dependency of the admin package.

* I excluded block editor packages in `product-editor`. This is because this project is bleeding edge WIP utilising block editor directly, and I don't want to cause issues for them as they are probably relying on latest APIs right now.

* I pinned `@types/wordpress__components` to try and best match the types of the `components` package in 6.0 as possible. 




### How to test the changes in this Pull Request:

Notes for testing/review:

1. A lot of the file changes here were automated by Syncpack, I don't expect you to check them all.
2. Probably the most valuable testing here would be a smoke test of the application. I would use the Live Branches feature to test this. (Please reach out if you need help enabling this).

When I say smoke test, I mean look for things that are obviously broken or a white screen type error in the basic WooCommerce screens, including onboarding and the WooCommerce home screen.




### Other information:




### FOR PR REVIEWER ONLY:
